pkgs/build-support/setup-hooks/wrap-apps-hook/wrap-apps-hook.sh

```diff
@@ -0,0 +1,79 @@
+# shellcheck shell=bash
+appsWrapperArgs=()
+
+# Inherit arguments given in mkDerivation
+appsWrapperArgs=( ${appsWrapperArgs-} )
+
+makeAppWrapper() {
+    local original="$1"
+    local wrapper="$2"
+    shift 2
+    makeWrapper "$original" "$wrapper" "${appsWrapperArgs[@]}" "$@"
+}
+
+wrapApp() {
+    local program="$1"
+    shift 1
+    wrapProgram "$program" "${appsWrapperArgs[@]}" "$@"
+}
+
+wrapAppsHook() {
+    # skip this hook when requested
+    [ -z "${dontWrapApps-}" ] || return 0
+
+    # guard against running multiple times (e.g. due to propagation)
+    [ -z "$wrapAppsHookHasRun" ] || return 0
+    wrapAppsHookHasRun=1
+
+    local targetDirs =( "$prefix/bin" "$prefix/sbin" "$prefix/libexec"  )
```

Comment:
Each of the hooks had a different set of `targetDirs`. I chose to consider all of them.